In [4]:
df = read.csv('https://raw.githubusercontent.com/Fabimath/CodJup/main/datos.csv')
reg=lm(Y~X1+X2+X3+X4,data=df)
summary(reg)


Call:
lm(formula = Y ~ X1 + X2 + X3 + X4, data = df)

Residuals:
     Min       1Q   Median       3Q      Max 
-21.7096  -3.5763   0.6177   5.2862  16.2480 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   8.1193    14.4766   0.561  0.57768    
X1           -0.3816     1.2255  -0.311  0.75692    
X2            0.5293     0.6829   0.775  0.44232    
X3            1.5950     0.2021   7.892 4.95e-10 ***
X4            0.9606     0.2751   3.492  0.00109 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 8.303 on 45 degrees of freedom
Multiple R-squared:  0.6352,	Adjusted R-squared:  0.6028 
F-statistic: 19.59 on 4 and 45 DF,  p-value: 2.136e-09


Significancia 0.02
1. ¿Se cumple el supuesto de independecia de los residuos?
2. ¿Qué se puede decir de la validez del modelo de regresión? Justifique mediante un test-F
3. ¿Comó se interpresta el R2?,¿y el R2 ajustado?, opine sobre la calidad del modelo.
4. ¿Qué covariable es mas significativa en el modelo?,¿por qué?, ¿que significa que sea 'más significativa'?
5. Opine sobre el valor del coeficiente de X1, ¿es razonable que valga 0?, ¿qué significa que valga 0 en el contexto?
6. Opine sobre el valor del coeficiente de X2, ¿es razonable que valga 0?, y con una significancia del 0.01, ¿es razonable que valga 0?
7. Considere este nuevo modelo:
   
     Y=β0+β3X3+β4X4
   
cual de los dos modelos trabajaría, ¿por qué?

1. P1

In [ ]:
shapiro.test(residuals(reg))


	Shapiro-Wilk normality test

data:  residuals(reg)
W = 0.96469, p-value = 0.1398


NO RECHAZO. SON NORMALES.

2. P2

`F-statistic: 19.59 on 4 and 45 DF,  p-value: 2.136e-09`

RECHAZO EL HECHO DE LOS COEFICIENTES SEAN 0. EL MODELO ES SIGNIFICATIVO.

3. P3

`Multiple R-squared:  0.6352,	Adjusted R-squared:  0.6028`

EL 63.52% DE LA VARIABILIDAD DE Y PUEDE SER EXPLICADA POR X1,X2,X3,X4. EL R2 AJUSTADO PENALIZA EL HECHO QUE ESTEN PRESENTES LAS 4 COVARIABLES.

EL MODELO ES RAZONABLE PERO NO ES DE MUY BUENA CALIDAD PUES EL R2 NO ES MUY CERCANO A 1.

4. P4

`X3            1.5950     0.2021   7.892 4.95e-10 ***`

X3 es más significativo pues tiene valor P más pequeño (4.95e-10)

5. P5

`X1           -0.3816     1.2255  -0.311  0.75692    `

SI PUES SU P-VALOR (0.75692) ES MAYOR A NUESTRA SIGNIFICANCIA (0.02). EL COEFICIENTE QUE ACOMPAÑA A X1 ES RAZONABLE QUE VALGA 0. INTERPRETACIÓN: X1 NO APORTA A LA EXPLIACIÓN DE Y.

6. P6

`X2            0.5293     0.6829   0.775  0.44232    `

SI PUES SU P-VALOR (0.44232) ES MAYOR A NUESTRA SIGNIFICANCIA (0.02) INCLUSIVE SI LA SIGNIFICANCIA FUERA 0.01  . EL COEFICIENTE QUE ACOMPAÑA A X2 ES RAZONABLE QUE VALGA 0. INTERPRETACIÓN: X2 NO APORTA A LA EXPLIACIÓN DE Y.

7. P7

In [8]:
reg2=lm(Y~X3+X4,data=df)
summary(reg2)


Call:
lm(formula = Y ~ X3 + X4, data = df)

Residuals:
     Min       1Q   Median       3Q      Max 
-21.5234  -3.7356   0.6367   5.5406  14.8021 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   9.1599     4.3032   2.129 0.038554 *  
X3            1.6323     0.1939   8.418 6.11e-11 ***
X4            0.9893     0.2680   3.692 0.000579 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 8.187 on 47 degrees of freedom
Multiple R-squared:  0.6296,	Adjusted R-squared:  0.6138 
F-statistic: 39.95 on 2 and 47 DF,  p-value: 7.307e-11


ESTE SEGUNDO MODELO TIENE TODAS SUS COVARIABLES SIGNIFICATIVAS Y SU R2 AJUSTADO ES MEJOR QUE EL DEL MODELO ANTERIOR. AUNQUE TIENE UN PEOR R2 ES TAN BAJA LA DIFERENCIA QUE POR EL HECHO DE QUE ESTE NUEVO MODELO TIENE DOS CARACTERIZTICAS MEJORES QUE EL ANTERIOR (R2 AJUSTADO Y SIGNIFICANCIA DE VARIABLES) ENTONCES SE SUGIERE TRABAJAR CON ESTE NUEVO MODELO.